### DISCUSSION TITLE
How to get order tracking number using SPAPI

### DISCUSSION THREAD

**Discussion by lowzijian** [amzn/selling-partner-api-models#3089] [General] [answered]
I'm integrating amazon marketplace with a private third party application and come across a dilemma when I found out that the **get orders** API does not return any shipment details ( carrier service id ,carrier service name , tracking number) for seller fulfilled orders (MFN) .In fact, we are not using any amazon shipping service , where seller required manually enter the tracking number while performing confirm shipping. 

I tried looked up for some related articles with keyword "get orders sp api with tracking number" and found out this https://sellercentral.amazon.com/forums/t/order-tracking-number-how-to-get-order-tracking-number-using-sp-api/979628
https://sellercentral.amazon.com/forums/t/api-call-to-get-tracking-information-for-a-order-item-for-fba-order/390981/6

I tried to get the orders by using GET_FLAT_FILE_ALL_ORDERS_DATA_BY_LAST_UPDATE_GENERAL using reports API , but it kept getting cancelled. I was clueless as there were no further instructions on what kind of information needed in request body from start of creating a new report. 

At the end , it is not an ideal solution where the report doesn't provide customer identifying information , which get orders with RDT has it. Besides , it seems tracking number / shipping carrier is not being returned in the report according to https://sellerstoolbox.com/reportTypes.jsp [MIGHT BE INACCURATE]

I need help.[😢](https://emojipedia.org/crying-face/)

**Comment by lowzijian** [comment]
Quick update: able to retrieve GET_FLAT_FILE_ALL_ORDERS_DATA_BY_LAST_UPDATE_GENERAL  report with proper dataStartTime & dataEndTime. However, it doesn't return any tracking number nor shipping carrier services.

**Comment by mehboobafridi** [comment]
Yes I am also suffering from this issue. Can anyone help in this regard

**Comment by nhtahoe** [comment]
Anyone figure out how to do this? Not seeing tracking info from any of the Orders apis or Reports apis

**Comment by mehboobafridi** [comment]
It is still an issue

**Comment by himanshusahu20** [comment]
@lowzijian @mehboobafridi @nhtahoe @hyandell @ferrants You all can use following reportTye to get tracking number for AFN orders,

{
    "reportType":"GET_AMAZON_FULFILLED_SHIPMENTS_DATA_GENERAL",
    "marketplaceIds":["{{MARKETPLACE_ID}}"],
    "dataStartTime":"2024-01-01T14:32:16.000Z",
    "dataEndTime":"2024-01-30T14:32:16.000Z"
}

This has worked for me.

**Reply by mehboobafridi** [reply]
@lowzijian has asked for MFN orders, not AFN.
Thanks for your contribution.

**Comment by NathanSmeltzer** [comment]
@lowzijian  do you have the [Direct to Consumer Shipping restricted role](https://developer-docs.amazon.com/sp-api/docs/direct-to-consumer-shipping-restricted-role) enabled for your developer profile? 

I'm waiting for my developer role update request to be completed by Amazon support. It can supposedly take up to 3 weeks or more for approval. I'm going to try all the restricted endpoints/reports if I'm ever approved.

This role opens up several additional reports and endpoints. However, I still do not see where tracking numbers are provided with these restricted endpoints and reports without creating the same shipments through the API.

In summary, I think it's not possible to get MFN shipments unless you create the shipments yourself through the API. I think this is how SaaS products like ShipStation get MFN shipments.

**Reply by tom-it** [reply]
any luck with getting tracking numbers from anywhere? I too have PII access, but am not able to find any report with tracking numbers

**Comment by agvmedia** [comment]
Has anyone been able to get tracking info from FBM orders?

**Comment by nbaybara** [comment]
Still looking for an answer, I have created a ticket lets see if it will be solved

**Reply by nbaybara** [reply]
Well apparently via API this is not possible, I wıll try to make ıntegratıon to the DHL to get ıt somehow

**Comment by tom-it** [comment]
I need this :( why is there no way to find tracking number

**Comment by chrisskhoury** [comment]
Hello :)

You can retrieve the tracking ID using the [Shipping v2 APIs](https://developer-docs.amazon.com/amazon-shipping/docs/shipping-api-v2-reference). The [getTracking](https://developer-docs.amazon.com/amazon-shipping/docs/shipping-api-v2-reference#gettracking) endpoint allows you to get live tracking updates. However, this only applies to shipments purchased through **Amazon Buy Shipping**. When you use the [purchaseShipment](https://developer-docs.amazon.com/amazon-shipping/docs/shipping-api-v2-reference#purchaseshipment) operation, it returns a tracking ID, which you can store and later use to get tracking information.

If you purchase shipping labels outside of Amazon Buy Shipping, you'll need to manually submit tracking information to Amazon using the [confirmShipment](https://developer-docs.amazon.com/sp-api/docs/orders-api-v0-reference#confirmshipment) operation to ensure a seamless buyer experience. This step is not required when using Amazon Buy Shipping, as tracking data is automatically handled within Amazon’s systems. But as of today, there is no API that returns tracking information for non Buy Shipping Labels.

Hope this helps clarify things!

Best,
Chris
Solution Architect Amazon SP API

**Reply by tom-it** [reply]
Hi Chris, thanks for your reply. 

The issue here then is, the merchant is buying amazon shipping labels through the amazon interface, and we need to somehow find the tracking number. 

but we only have the order data (through SP API). could we get the tracking number with an order number from the shipping V2 apis? so we can match shipments from Amazon Buy Shipping with our orders?

**Reply by agvmedia** [reply]
@chrisskhoury: Is it possible for you to submit a feature request to expose the tracking ID and carrier in the Orders API for all orders, not just **Amazon Buy Shipping** orders? 
This information is already available in the Orders view on Seller Central, so I would assume it could be included in the API as well.

**Reply by ritikacd** [reply]
Hi @tom-it,

Currently, tracking details are not available in SP-APIs if seller uses Amazon directly. As mentioned [here](https://github.com/amzn/selling-partner-api-models/discussions/3089#discussioncomment-12761149), if developer uses the [purchaseShipment](https://developer-docs.amazon.com/amazon-shipping/docs/shipping-api-v2-reference#purchaseshipment), it returns a tracking ID, which you can store and later use to get tracking information. 

Thank you for providing a valuable feedback, we have captured it. 

Thanks,
Ritika
Selling Partner API Developer Services

**Comment by MetaMan2023** [comment]
I know this is not related to this question -- but I am having trouble with "purchaseShipment" where I keep receiving this error:

[An invalid documentation specification was provided. Please provide a documentation specification based on the GetRates response.](https://github.com/amzn/selling-partner-api-models/issues/4903)

Has anyone else had this issue?

**Comment by Quankzx** [comment]
Hello, I’m also facing a similar issue with retrieving the tracking ID for MFN orders. Does anyone have a feasible solution for this yet?

**Comment by chrisskhoury** [comment, marked as answer]
Hello! I have some good news :) 

The Orders API v2026-01-01 is now officially launched, and we've added comprehensive package tracking information specifically for FBM orders based on feedback like yours!

You can now retrieve tracking IDs, carrier information, and package status directly through the new packages[] array. Simply use the getOrder or searchOrders endpoints with includedData=PACKAGES:

`GET /orders/2026-01-01/orders/{orderId}?includedData=PACKAGES`

This works for both new orders and historical orders, so you can backfill any missing tracking data.

Check out the official documentation for complete details and examples:

•	[Orders API v2026-01-01 Announcement](https://developer-docs.amazon.com/sp-api/changelog) - Official launch announcement and release notes
•	[Orders API Reference](https://developer-docs.amazon.com/sp-api/docs/orders-api-v2026-01-01-reference) - Complete API specification with endpoints, parameters, and schemas
•	[Orders API Migration Guide](https://developer-docs.amazon.com/sp-api/docs/orders-api-v2026-01-01-migration-guide) - Comprehensive step-by-step migration documentation
•	[Orders API Use Case Guides](https://developer-docs.amazon.com/sp-api/docs/orders-api-v2026-01-01-use-cases) - Implementation patterns for common scenarios
•	[Migrate to Orders API v2026-01-01: Interactive Developer Guide](https://github.com/amzn/selling-partner-api-samples/discussions/253)
•	[Postman Collection: Orders API Migration Collection (Interactive hands-on migration guide)](https://www.postman.com/amazon-selling-partner-api/sp-api/collection/45158253-23edb9a3-8a58-478f-82b8-3f43805f17a4?action=share&source=copy-link&creator=45158253)[Use Case Guides](https://developer-docs.amazon.com/sp-api/docs/orders-api-v2026-01-01-use-cases) - Real-world examples

Questions?
📧 [orders-api-feedback@amazon.com](mailto:orders-api-feedback@amazon.com) or 💬 [GitHub Discussions](https://github.com/amzn/selling-partner-api-samples/discussions)

This should solve your tracking ID challenge! Let us know how it goes. 🚀

Best,

Chris Khoury
Solution Architect - Amazon SP-API

**Reply by agvmedia** [reply]
Legend!
